src/main/java/org/opensearch/knn/index/mapper/EngineFieldMapper.java

```diff
@@ -0,0 +1,285 @@
+                    : VectorEncoding.BYTE;
+                fieldType.setVectorAttributes(
+                    adjustedDimension,
+                    encoding,
+                    SpaceType.DEFAULT.getKnnVectorSimilarityFunction().getVectorSimilarityFunction()
+                );
+            } else {
+                fieldType.setDocValuesType(DocValuesType.BINARY);
+            }
+
+            this.fieldType.freeze();
+            this.vectorTransformer = knnLibraryIndexingContext.getVectorTransformer();
+        }
+
+        // Common Attributes
+        this.perDimensionProcessor = knnLibraryIndexingContext.getPerDimensionProcessor();
+        this.perDimensionValidator = knnLibraryIndexingContext.getPerDimensionValidator();
+        this.vectorValidator = knnLibraryIndexingContext.getVectorValidator();
+    }
+
+    @Override
+    protected List<Field> getFieldsForFloatVector(final float[] array, boolean isDerivedSourceEnabled) {
+        if (this.isLuceneEngine) {
+            final List<Field> fields = new ArrayList<>();
+            fields.add(new DerivedKnnFloatVectorField(name(), array, fieldType, isDerivedSourceEnabled));
+            if (hasDocValues && vectorFieldType != null) {
+                fields.add(new VectorField(name(), array, vectorFieldType));
+            }
+            if (stored) {
+                fields.add(createStoredFieldForFloatVector(name(), array));
+            }
+            return fields;
+        }
+        return super.getFieldsForFloatVector(array, isDerivedSourceEnabled);
+    }
+
+    @Override
+    protected List<Field> getFieldsForByteVector(final byte[] array, boolean isDerivedSourceEnabled) {
+        if (this.isLuceneEngine) {
+            final List<Field> fields = new ArrayList<>();
+            fields.add(new DerivedKnnByteVectorField(name(), array, fieldType, isDerivedSourceEnabled));
+            if (hasDocValues && vectorFieldType != null) {
+                fields.add(new VectorField(name(), array, vectorFieldType));
+            }
+            if (stored) {
+                fields.add(createStoredFieldForByteVector(name(), array));
+            }
+            return fields;
+        }
+        return super.getFieldsForByteVector(array, isDerivedSourceEnabled);
+    }
+
+    @Override
+    protected VectorValidator getVectorValidator() {
+        return vectorValidator;
+    }
+
+    @Override
+    protected PerDimensionValidator getPerDimensionValidator() {
+        return perDimensionValidator;
+    }
+
+    @Override
+    protected PerDimensionProcessor getPerDimensionProcessor() {
+        return perDimensionProcessor;
+    }
+
+    @Override
+    protected VectorTransformer getVectorTransformer() {
+        if (isLuceneEngine) {
+            return super.getVectorTransformer();
+        }
+        return vectorTransformer;
+    }
+
+    @Override
+    void updateEngineStats() {
+        if (this.isLuceneEngine) {
+            KNNEngine.LUCENE.setInitialized(true);
```

Comment:
Can we just initialize all of them?

Reply:
Do you mean based on the mapper engine or just generally set all to true?

Reply:
Like engine.setInitialized to true

Reply:
Updated.